cmd/refresh-uids-from-ferry/main.go

```diff
@@ -357,37 +380,56 @@ func run(ctx context.Context) error {
 		tracing.LogErrorWithTrace(span, exeLogger, msg)
 		// Start up a notification manager JUST for the purpose of sending the email that we couldn't open the DB.
 		// In the case of this executable, that's a fatal error and we should stop execution.
-		admNotMgr, aReceiveChan, adminNotifications := setupAdminNotifications(ctx, nil)
-		sendSetupErrorToAdminMgr(aReceiveChan, msg)
-		close(aReceiveChan)
-
-		if err2 := sendAdminNotifications(ctx, admNotMgr, &adminNotifications); err2 != nil {
-			msg := "error sending admin notifications"
-			tracing.LogErrorWithTrace(span, exeLogger, msg)
-			err := fmt.Errorf("%s regarding %w: %w", msg, err, err2)
-			return err
+		if !viper.GetBool("disableNotifications") {
+			admNotMgr, aReceiveChan, adminNotifications := setupAdminNotifications(ctx, nil)
+			sendSetupErrorToAdminMgr(aReceiveChan, msg)
+			close(aReceiveChan)
+
+			// We need to pass the pointer in here since we need to pick up the
+			// changes made to adminNotifications, as explained here:
+			// https://stackoverflow.com/a/52070387
+			// and mocked out here:
+			// https://go.dev/play/p/rww0ORt94pU
+			if err2 := sendAdminNotifications(ctx, admNotMgr, &adminNotifications); err2 != nil {
+				msg := "error sending admin notifications"
+				tracing.LogErrorWithTrace(span, exeLogger, msg)
+				err := fmt.Errorf("%s regarding %w: %w", msg, err, err2)
+				return err
+			}
 		}
 		return fmt.Errorf("%s: %w", msg, err)
 	}
 	defer database.Close()
 	span.AddEvent("Opened ManagedTokensDatabase")
 
 	// Send admin notifications at end of run
-	admNotMgr, aReceiveChan, adminNotifications := setupAdminNotifications(ctx, database)
+	// Note:  We don't actually need the values of admNotMgr and adminNotifications beyond the next if-clause.  However, we will need
+	// the value of aReceiveChan, and since setupAdminNotifications returns all three, we want to initialize all three, so they can possibly
+	// be overridden in the next if-clause.  If they are not overridden, they will all stay nil, which is fine.
+	var admNotMgr *notifications.AdminNotificationManager
+	var aReceiveChan chan<- notifications.SourceNotification // Channel to send admin notifications on.  We need it here so we can close it at the end of the run
```

Comment:
so aReceiveChan is the channel you are sending to or actually receiving from? Was just a tad confused based on this comment if the name.

Reply:
Yeah - naming things is tough, so I understand the confusion.  `aReceiveChan` is short for `Admin Notifications Receive Channel` in my mind - so basically, the channel that `notifications.AdminNotificationsManager` will _receive_ messages from.

The sender in this case is the main function in this file.

There's also an explicit hint in the code:
```go
	var aReceiveChan chan<- notifications.SourceNotification
```

Note the type here is `chan<- notifications.SourceNotification`.  The `chan<-` bit is Go syntax that says "the type of this variable is a channel that can only be sent on".  Similarly, `<-chan` means a channel that can only be received on.  `chan` means  a channel you can both send to and receive from.  This is very powerful, because it means you can control who the possible senders and receivers are for a certain channel.

In the case of _this_ code, `aReceiveChan` is declared this way because if we are sending notifications (`if !disableNotifications`), then the call to [setupAdminNotifications](https://github.com/fermitools/managed-tokens/blob/issue-98/cmd/refresh-uids-from-ferry/utils.go) returns as its second value a `chan<- notifications.SourceNotification`, so we need to make sure our initialization matches the variable type.  

Last note: `setupAdminNotifications` itself is returning that type to signal to the callers that they can only use the returned channel to send values on.